### PR TITLE
GEODE-8475: Resolve a potential dead lock in ParallelGatewaySenderQueue

### PR DESCRIPTION
**Pull request by gesterzhou**
Co-authored-by: Darrel Schneider <darrel@vmware.com>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java [hunk 1]
```diff
@@ -755,12 +755,16 @@ public boolean put(Object object) throws InterruptedException, CacheException {
                 bucketFullPath, brq);
           }
           if (brq != null) {
+            boolean intializingLocked = brq.lockWhenRegionIsInitializing();
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:758]: Can we add unit tests to make sure failed Initialization lock is held during put.

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java [hunk 2]
```diff
@@ -755,12 +755,16 @@ public boolean put(Object object) throws InterruptedException, CacheException {
...
             brq.getInitializationLock().readLock().lock();
             try {
               putIntoBucketRegionQueue(brq, key, value);
               putDone = true;
             } finally {
               brq.getInitializationLock().readLock().unlock();
+              if (intializingLocked) {
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:765]: I assume we don't have to worry about the above unlock code throwing any exception...

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java (hunk 1 above)
Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:758]: added.

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java (hunk 2 above)
Comment (gesterzhou) [reply geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:765]: yes

**Pull request merged**